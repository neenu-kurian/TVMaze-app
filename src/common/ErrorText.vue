<template>
  <p class="error__message">
    <slot>Sorry, we couldnt find any results. Please try another tv show</slot>
  </p>
</template>

<script>
export default {
  name: 'ErrorText'
}
</script>
