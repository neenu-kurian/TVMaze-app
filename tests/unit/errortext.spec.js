import { mount } from '@vue/test-utils'
import ErrorText from '@/common/ErrorText'

describe('ErrorText', () => {
  it('displays message correctly', () => {
    const wrapper = mount(ErrorText)

    expect(wrapper.text()).toContain(
      'Sorry, we couldnt find any results. Please try another tv show'
    )
  })

  it('displays slot text correctly', () => {
    const wrapper = mount(ErrorText, {
      slots: {
        default: 'Something went wrong'
      }
    })

    expect(wrapper.text()).toContain(
      'Something went wrong'
    )
  })

  it('renders component correctly', () => {
    const wrapper = mount(ErrorText)

    expect(wrapper).toMatchSnapshot()
  })
})
