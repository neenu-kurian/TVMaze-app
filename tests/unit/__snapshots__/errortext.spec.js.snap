// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorText renders component correctly 1`] = `<p class="error__message">Sorry, we couldnt find any results. Please try another tv show</p>`;

exports[`Errortext renders component correctly 1`] = `<p class="error__message">Sorry, we couldnt find any results. Please try another tv show</p>`;

exports[`errortext renders component correctly 1`] = `<p class="error__message">Sorry, we couldnt find any results. Please try another tv show</p>`;

exports[`renders component correctly 1`] = `<p class="error__message">Sorry, we couldnt find any results. Please try another tv show</p>`;
